### PR TITLE
PROGRAMMES-6962: Category listings

### PR DESCRIPTION
**Pull request by BjornTwachtmann**
See also https://github.com/bbc/programmes-pages-service/pull/287

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/ArgumentResolver/CategoryByUrlKeyValueResolver.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    {
+        $this->categoriesService = $categoriesService;
+    }
+
+    public function supports(Request $request, ArgumentMetadata $argument): bool
+    {
+        return (
+            $request->attributes->has('categoryHierarchy')
+            && $request->attributes->has('categoryType')
+            && in_array($argument->getType(), self::SUPPORTED_CLASSES)
```
Comment (Tasiobg) [inline src/ArgumentResolver/CategoryByUrlKeyValueResolver.php]:
```suggestion
            && \in_array($argument->getType(), self::SUPPORTED_CLASSES)
```
Looks weird adding a backslash but apparently it helps a bit with opcode optimizations https://github.com/Roave/FunctionFQNReplacer
Not a required changed, mostly a peculiarity I wanted to share about how PHP works

File: src/Ds2013/Presenters/Section/Category/Sidenav/CategorySidenavPresenter.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        $this->categoryType = $categoryType;
+        $this->router = $router;
+    }
+
+    public function getChildren(): array
+    {
+        return $this->category->getChildren();
+    }
+
+    public function getSliceLink(string $slice)
```
Comment (Tasiobg) [inline src/Ds2013/Presenters/Section/Category/Sidenav/CategorySidenavPresenter.php]:
```suggestion
    public function getSliceLink(string $slice): string
```

File: src/Ds2013/Presenters/Section/Category/Sidenav/CategorySidenavPresenter.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        return $this->router->generate('category_slice', [
+            'categoryType' => $this->categoryType,
+            'categoryHierarchy' => $this->category->getUrlKeyHierarchy(),
+            'slice' => $slice,
+        ]);
+    }
+
+    public function getChildLink(Category $child)
```
Comment (Tasiobg) [inline src/Ds2013/Presenters/Section/Category/Sidenav/CategorySidenavPresenter.php]:
```suggestion
    public function getChildLink(Category $child): string
```

File: src/Controller/Category/ListingController.php [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    private const RESULTS_PER_PAGE = 50;
+
+    public function __invoke(
+        string $slice,
+        string $categoryType,
+        Category $category,
+        CategoriesService $categoriesService,
+        ProgrammesService $programmesService
+    ) {
+        $this->setAtiContentLabels('communist', 'party');
```
Comment (Tasiobg) [inline src/Controller/Category/ListingController.php]: 🤔

Comment (BjornTwachtmann) [reply src/Controller/Category/ListingController.php]: There's another ticket to actually set thse

**Pull request merged**